### DISCUSSION TITLE
Can shed skin export generators from shed skin-built code to CPython code?

### DISCUSSION THREAD

**Discussion by dstromberg** [shedskin/shedskin#421] [Q&A]
I've got a module that works on CPython 3.11.  It has a couple of functions, and a couple of generators, and a class.

If I compile it with shedskin, then I can see the functions and the class in a dir(), but I don't see the generators.  No error is emitted for the yield commands.

Is shedskin able to export a generator from shedskin-compiled code to CPython code?  If no, is there a practical workaround?

**Comment by dstromberg** [comment]
If I call the generator from the Shedskin code to the Shedskin code, the yield seems to work.

**Reply by dstromberg** [reply]
Do I need a class with `__init__`,` __iter__` and `__next__`, instead of the yield-based style?

**Comment by dstromberg** [comment]
I tried:
```
class foo_iter():
    def __init__(self, n):
        self.n = n
        self.i = -1

    def __iter__(self):
        return self

    def __next__(self):
        self.i += 1
        if self.i < n:
            return self.n
        raise StopIteration
```

It showed up in dir(), but foo_iter didn't appear to have `__iter__` or `__next__` methods.

**Comment by dstromberg** [comment]
I tried calling foo_iter(5) from shedskin-compiled code, and got:
`*WARNING* rolling_checksum_ss_mod.py:26: '__iter__' is not supported
`
...so it's sounding like this isn't going to happen without changes to shedskin itself.

**Comment by srepmub** [comment]
hi dan! yeah, these are currently not exported, so you'll have to find a work-around. I don't know if I will ever get around to implementing this..

**Comment by srepmub** [comment]
older issue about this:

https://github.com/shedskin/shedskin/issues/133

**Comment by srepmub** [comment]
btw we probably need at least a warning when generating the extension module..
